streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -312,6 +316,10 @@ void setConsumer(final Consumer<byte[], byte[]> consumer) {
      * @throws TaskMigratedException if the task producer got fenced or consumer discovered changelog offset changes (EOS only)
      */
     boolean updateNewAndRestoringTasks() {
+        if (versionProbingFlag) {
```

Comment:
Do we actually need this check, since assignedActiveTasks and assignedStandbyTasks are empty, so active.created and standby.created will also be empty?

Reply:
Maybe not. I added it as "early return" path -- let me know if you think it's better to not add it.